src/plugins/pendo/agent/ajax.ts

```diff
@@ -0,0 +1,111 @@
+import { logger } from 'lib/logger'
+
+import { PendoAjaxError } from './errors'
+import { filterRequest, filterResponse } from './filters'
+import type { PendoEnv } from './types'
+
+const moduleLogger = logger.child({ namespace: ['Plugin', 'Pendo', 'Agent', 'Ajax'] })
+
+function getTransmissionData(this: PendoEnv, url: URL, body: BodyInit | undefined) {
```

Comment:
That doesn't look right, can we add a test for this? `Function.prototype.call()` calls a function with the first param used as a `this` for that function, and the rest `...args`. It doesn't mutate the function signature to make `this` available explicitly as a first parameter of said function.

Reply:
Just tried it and this works as intented because of the use of the `this` keyword:
https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAWwJ4DFzXkgFFACxgGcAaRAI3IgEpEBvAKEUQgWLgBsBTAOk7gBzXPUIBDKAC5EhEuSoQAvjUaLGjNJkiwEvCGM6cRi8gHJgcOKbMUxAJ1M0gA

Would be cleaner to avoid this magic, but if we need it for type purposes I'm fine with it